components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1317,6 +1359,13 @@ where
             SignificantMsg::UnsafeRecoveryFillOutReport(syncer) => {
                 self.on_unsafe_recovery_fill_out_report(syncer)
             }
+            // for snapshot recovery (safe recovery)
+            SignificantMsg::RecoveryLeaderWaitApply(syncer) => {
+                self.on_recovery_leader_wait_apply(syncer)
+            }
+            SignificantMsg::RecoveryFollowerWaitApply(syncer) => {
```

Comment:
didn't see anywhere send it

Reply:
Yes, you are right, the purpose of this PR is to provide these two interfaces.  so that we can use them two in late PR.

Reply:
hardly to say if it's appropriate without the implementation

Reply:
another PR is progressing, will submit it soon

Reply:
The message sender shouldn't know whether the peer is leader or not. And it should be a generic function instead of only for recovery. So I prefer to rename to `WaitApplyToLast`.

Reply:
it looks good ideas, it already merge two into one interface.

Reply:
> hard to say if it's appropriate without the implementation

https://github.com/tikv/tikv/pull/13028/files#diff-268217152449da0fc3bdaaabae08fba32656a98c17e29bf3d3491f25c2613166